cpp/oneapi/dal/algo/decision_forest/common.hpp

```diff
@@ -603,6 +606,15 @@ class descriptor : public detail::descriptor_base<Task> {
         base_t::set_seed_impl(value);
         return *this;
     }
+
+    oneapi::dal::backend::primitives::engine get_engine() const {
+        return base_t::get_engine();
+    }
+
+    auto& set_engine(oneapi::dal::backend::primitives::engine value) {
+        base_t::set_engine_impl(value);
+        return *this;
+    }
```

Comment:
You **should not** use internal functionality in public interfaces. This thing is twofold:
1) headers from `primitives` are not exposed to user at all,
2) this API can be changed at any moment, we have no commitment to preserve it 

Thus I have 2 proposals:
1) Move it to the `detail` section, but you need to move away from `primitives` entities anyway
2) You can create a separate `random_number_generator` algorithms for different types of RNGs. Please take a look on combination of `svm` + `rbf_kernel`, `linear_kernel`

Reply:
Darn. I think either option would probably require a review of architecture before implementation, and would impact all of dal... Unfortunately its beyond the scope/time allocation, I'll probably close.